src/checkers/inference/solver/util/PrintUtils.java

```diff
@@ -142,4 +147,133 @@ private static void buildStatisticText(String key, Integer value, StringBuilder
         statisticsText.append(value);
         statisticsText.append("\n");
     }
+
+    public static void printUnsolvable(Collection<Constraint> mus) {
+        if (mus == null) {
+            System.out.println("The backend you used doesn't support explanation feature!");
+            return;
+        }
+
+        ToStringSerializer toStringSerializer = new ToStringSerializer(false);
+        SlotPrinter slotPrinter = new SlotPrinter(toStringSerializer);
+        // Print constraints and related slots
+        System.out.println("\n=================================== Explanation Starts=================================\n");
```

Comment:
Use stringbuilder? A single println call is better as it would be 1 I/O interrupt.

Reply:
Sure.

Reply:
Separate PR (also for ToStringSerializer)